docs/docs/guides/developer-guide/cli/index.md

```diff
@@ -66,10 +75,99 @@ The CLI will guide you through the process of adding new functionality to your p
 The `add` command is much more than a simple file generator. It is able to
 analyze your project source code to deeply understand and correctly update your project files.
 
+### Non-Interactive Mode
+
+For automation or when you know exactly what you need to add, you can use the non-interactive mode with specific flags:
+
+<Tabs groupId="package-manager">
+<TabItem value="npm" label="npm" default>
+
+```bash
+# Create a new plugin
+npx vendure add -p MyPlugin
+
+# Add an entity (requires plugin context)
+npx vendure add -e MyEntity
+
+# Add a service (requires plugin context)  
+npx vendure add -s MyService
```

Comment:
What means `requires plugin context`. This is not clear for a developer at this point.